compiler/runtime/memory.c

```diff
@@ -161,14 +161,10 @@ int memcmp(const void* ptr1, const void* ptr2, size_t num) {
 
 void* sk_malloc(size_t size) {
   void* result = malloc(size);
-  if (result == NULL) {
-    perror("malloc");
-    exit(1);
-  }
   sk_lower_static(result);
 
   if (result == NULL) {
-    fprintf(stderr, "Out of memory\n");
+    perror("malloc");
     SKIP_throw_cruntime(ERROR_OUT_OF_MEMORY);
   }
```

Comment:
It could go either way. `sk_lower_static` won't change `result`, so the whole process will crash either way. I went with the current code since this error check is older than the one above. Happy to change the order though.

Reply:
I think doing the error check directly after malloc is clearer, so change it.